### ISSUE TITLE
Crowd control gun brainstorming

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#194]
The game desperately needs a gun which is powerful against multiple enemies. Can we agree on that? Good.

Let's think of how such a gun would work and look like.

A tesla coil chain lightning gun seems like an obvious choice. Bulletspam (likely bullets connected by lines to make it look like lightning) which, upon hitting an enemy, will chain to the nearest enemy, and again and again, dealing damage to each.

But i'm more of a sniper enthusiast, and so my proposal is inevitably going to be: a sniper gun, which, once it hits an enemy, explodes on that enemy creating a bubble. All the enemies within that bubble take the damage, 100% and +33% per each enemy inside of the bubble. So hitting 4 enemies with this gun would instakill all of them.

Do you have any better ideas? Please share. Please also discuss the above options. Which one do you prefer?

**Comment by Gregory-Y** [comment]
I had a vaguely similar idea for adding the XM25 grenade launcher which airburst rounds which deal like 40 area damage plus 20 damage on contact, not necessarily killing people but encouraging them to stay away from crowded areas like points. I liked this weapon idea because it could make a large area dangerous to enemies. 

Seeing your idea I'm thinking it would be better for the grenade to only explode when it hits an enemy like your sniper rifle idea since doing so requires more skill. As for damage, I think it should be changed to 25 base damage, plus 10-15 for every enemy within the explosion's aoe, though maybe it should be a little lower. For mag cap I was thinking 5 rounds, and a 1000-1500 ms fire rate. Since this weapon has the potential to be a menace on compound I also think the projectile should be visible through roofs.

What I wasn't so sure about was the projectile speed. On one hand, a weapon with the potential to get an easy quad kill should not fire a projectile that moves at sniper speed, because that would be a little too in**nova**tive, but if the projectile is too slow the weapon would be pretty much impossible to use. I think the optimal speed would be like 80 or something, but i'm really not sure.

What do you think of this idea? Also whether the sniper rifle or the grenade launcher is added the crowd control weapon should probably negate cover to make it more of a threat to groups.

**Comment by deferred-impact** [comment]
> but if the projectile is too slow the weapon would be pretty much impossible to use. I think the optimal speed would be like 80 or something, but i'm really not sure.

It seems clear that the 50 bullet speed rockets are very much possible to use. So 50 bullet speed would work just fine here.

> What do you think of this idea? Also whether the sniper rifle or the grenade launcher is added the crowd control weapon should probably negate cover to make it more of a threat to groups.

Well it's pretty much exactly my explosive sniper idea, except with less bullet speed i guess, so of course i am in favor of it.

What would be the range of the explosion? I suppose it would need to be quite large, so that the benefit of hitting multiple enemies is easier to achieve

**Comment by Gregory-Y** [comment]
This is kinda bad but about this big 
![area](https://github.com/deferred-impact/battledudes-issue-tracker/assets/68972578/33dbf913-024a-4596-b6e0-2c2e61dc1ceb)

**Comment by deferred-impact** [comment]
Hm. A compelling size. Though maybe slightly larger would be best.

**Comment by Haox09** [comment]
make it the lowest firerate gun

**Comment by Gregory-Y** [comment]
1000-1500ms would be the lowest fire rate probably and yes it should be low since it has a lot of destructive potential.

**Comment by deferred-impact** [comment]
Low firerate seems like a perfectly fair price to pay for absolutely demolishing whole teams

**Comment by deferred-impact** [comment]
Folks in discord brought up concerns about spamming this thing onto Hardpoint, which is definitely a valid concern. Perhaps this thing should be blocked by the Trophy just like the explosives are.

**Comment by Gregory-Y** [comment]
Just want to add that if the player gets caught in the bubble they should take the same amount of damage as all the enemy players so that they don't spam this weapon at close range.

**Comment by deferred-impact** [comment]
Seems logical to me.

**Comment by deferred-impact** [comment]
A very valid thing brought up by @NOTGregory ,

> [...] on maps with few obstacles like westlands and parts of campsite [...] on maps with little cover those groups of enemies are more problematic since they can lay down overlapping fields of fire which are more difficult to dodge, and I don't know how to account for this either.

Crossfire is a problem, perhaps an even bigger problem than amassed fire. Ideally, we should be designing crowd control guns with this in mind. The best crowd control gun is one which works just as well no matter how the enemies are positioned on your screen; its scaling only cares about the total amount of enemies threatening you.
